src/libraries/Common/src/System/Net/IPv6AddressHelper.Common.cs

```diff
@@ -165,18 +185,21 @@ internal static unsafe bool IsValidStrict(char* name, int start, ref int end)
 
                             // If there's more after the closing bracket, it must be a port.
                             // We don't use the port, but we still validate it.
-                            if (i + 1 < end && name[i + 1] != ':')
+                            if (i + 1 < end && name[i + 1] != TChar.CreateTruncating(':'))
                             {
                                 return false;
                             }
 
                             // If there is a port, it must either be a hexadecimal or decimal number.
-                            if (i + 3 < end && name[i + 2] == '0' && name[i + 3] == 'x')
+                            // If the next two characters are '0x' or '0X' then it's a hexadecimal number. Skip the prefix.
+                            if (i + 3 < end && name[i + 2] == TChar.CreateTruncating('0') && (name[i + 3] == TChar.CreateTruncating('x') || name[i + 3] == TChar.CreateTruncating('X')))
                             {
                                 i += 4;
                                 for (; i < end; i++)
                                 {
-                                    if (!char.IsAsciiHexDigit(name[i]))
+                                    int ch = int.CreateTruncating(name[i]);
+
+                                    if (ch is not (>= '0' and <= '9' or >= 'a' and <= 'f' or >= 'A' and <= 'F'))
```

Comment:
Is this correct from a precedence perspective? If so, it's very non-obvious. Please add parens.